### ISSUE TITLE
create a feature for calculating volume within a given cord extension

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#92]
possible features:
- per vertebral level
- per slice range

bit overlap with sct_extract_metric --> use code instead of duplicating (see with Simon). Maybe create a module or so.

**Comment by jcohenadad** [comment]
New idea:
call sct_extract_metric in sct_process_segmentation to compute CSA and volume within given vertebral levels or slices.

**Comment by jcohenadad** [comment]
Hi Simon,

I tested sct_process_segmentation in your branch. I noticed a problem:

data: sct_example_data

syntax:
~~~~
sct_process_segmentation -i t2_seg.nii.gz -p compute_csa -z 20:21
~~~~

the result is:
0,  mean CSA:    31.8622341156 +/- 0.0

however, the CSA at 20 and 21 is:
z=20: 33.0007480349 mm^2
z=21: 32.9382462251 mm^2

so there is a problem...

**Comment by slevyrosetti** [comment]
Ok, I'm on it.

Simon
Le 2015-03-22 16:46, "Julien Cohen-Adad" <notifications@github.com> a écrit
:

**Comment by slevyrosetti** [comment]
Bug came from discrepancy between CSA results displayed in terminal window and those written in files csa_volume.nii.gz and csa.txt. I fixed it in commit #4d639e9.

**Comment by jcohenadad** [comment]
another issue:

~~~~
sct_process_segmentation -i t2_seg.nii.gz -p compute_csa -z 20
~~~~

ERROR:
~~~~
Traceback (most recent call last):
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_extract_metric", line 885, in <module>
    main()
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_extract_metric", line 272, in main
    data = remove_slices(data, slices_of_interest)
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_extract_metric", line 535, in remove_slices
    slices_list = [i for i in range(slices_range[0], slices_range[1]+1)]
IndexError: list index out of range
~~~~

**Comment by slevyrosetti** [comment]
Issue came from sct_extract_metric that didn't allow to select one slice. It is now fixed in commit #ad63261 on branch simon_process_seg.

**Comment by slevyrosetti** [comment]
Not exactly fixed actually, I'm on it.

**Comment by slevyrosetti** [comment]
Now completely fixed in commit #2cfa57e on branch simon_process_seg.
